pkg/templates/pdbminavailable/template.go

```diff
@@ -0,0 +1,178 @@
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/pdbminavailable/internal/params"
+	pdbV1 "k8s.io/api/policy/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	templateKey = "pdb-min-available"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Pod Disruption Budget Configuration Checks",
+		Key:         templateKey,
+		Description: "Flag non-optimal PDB configurations",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{
+				objectkinds.PodDisruptionBudget},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return func(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+
+				var results []diagnostic.Diagnostic
+
+				// Get the PDB provided
+				pdb, ok := object.K8sObject.(*pdbV1.PodDisruptionBudget)
+				if !ok {
+					return nil
+				}
+
+				// Extract the MinAvailable value from the spec and check if it's a number or percentage
+				value, isPercent, err := getIntOrPercentValueSafelyFromString(pdb.Spec.MinAvailable.String())
+				if err != nil {
+					return []diagnostic.Diagnostic{
+						{
+							Message: fmt.Sprintf("pdb has invalid MinAvailable value: %v", err),
+						},
+					}
+				}
+
+				// If the value is a percentage, handle the case where the MinValue is set to 100%
+				// as DeploymentLike's replica counts don't need to be compared
+				if isPercent {
+					if value == 100 {
+						return []diagnostic.Diagnostic{
+							{
+								Message: "PDB has minimum available replicas set to 100 percent of replicas",
+							},
+						}
+					}
+				}
+
+				// Build the label selector for the PDB to use for comparison
+				labelSelector, err := metaV1.LabelSelectorAsSelector(&metaV1.LabelSelector{
+					MatchLabels:      pdb.Spec.Selector.MatchLabels,
+					MatchExpressions: pdb.Spec.Selector.MatchExpressions},
+				)
+
+				if err != nil {
+					return []diagnostic.Diagnostic{
+						{
+							Message: fmt.Sprintf("pdb has invalid label selector: %v", err),
+						},
+					}
+				}
+
+				// Evaluate Deploymet Likes in the lintContext to see if they have MinAvailable set too low
+				deploymentLikes, err := getDeploymentLikeObjects(lintCtx, labelSelector, pdb.Namespace)
+				if err != nil {
+					return []diagnostic.Diagnostic{
+						{
+							Message: fmt.Sprintf("Failed to retrieve deployments matching the PDB's label selector within namespace %s: %v", pdb.Namespace, err.Error()),
+						},
+					}
+				}
+
+				for _, dl := range deploymentLikes {
+					pdbMinAvailable := value
+					replicas, _ := extract.Replicas(dl)
+					if isPercent {
+						// Calulate the actual value of the MinAvailable with respect to the Replica count if a percentage is set
+						pdbMinAvailable = int(replicas) * (value / 100)
```

Comment:
Won't it be always zero? I guess you need something like  `pdbMinAvailable = float64(replicas) * (float64(value) / float64(100))`

Reply:
In the end this still has to be an integer. It doesn't matter too much though because as long as it's less than 100%, it should be a valid config. For accuracy though, I think this will work:
`pdbMinAvailable = int(math.Floor(float64(replicas) * (float64(value) / float64(100))))`

This finds the exact value of minavailable based on a percentage, and then rounds down so cases like 1 replica with a 50% minavaiable (which calculates to 0.5) won't cause this remediation to fire. I'm not sure exactly how kubernetes handles this sort of case, so I'm erring on the side of permissive at first. It might be a good place for iteration

Reply:
@tremes is right but if we change order (multiplication first) it will work without conversion and floor

```
pdbMinAvailable = (int(replicas) * value) / 100
```